front/src/applications/stdcmV2/views/StdcmViewV2.tsx

```diff
@@ -5,60 +5,138 @@ import { useTranslation } from 'react-i18next';
 import { useSelector } from 'react-redux';
 
 import useStdcm from 'applications/stdcm/hooks/useStdcm';
-import { useOsrdConfSelectors } from 'common/osrdContext';
+import { useOsrdConfSelectors, useOsrdConfActions } from 'common/osrdContext';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import { useAppDispatch } from 'store';
 
 import StdcmConfig from '../components/StdcmConfig';
 import StdcmHeader from '../components/StdcmHeader';
 import StdcmResults from '../components/StdcmResults';
-import type { StdcmSimulationResult } from '../types';
+import type { StdcmSimulation, StdcmSimulationInputs } from '../types';
 
 const StdcmViewV2 = () => {
   const { t } = useTranslation('stdcm');
-  const { launchStdcmRequest, cancelStdcmRequest, isPending, stdcmV2Results, pathProperties } =
-    useStdcm();
+  const {
+    launchStdcmRequest,
+    cancelStdcmRequest,
+    isPending,
+    isSuccessful,
+    isStdcmResultsEmpty,
+    stdcmV2Results,
+    pathProperties,
+  } = useStdcm();
   const { getScenarioID } = useOsrdConfSelectors();
   const scenarioID = useSelector(getScenarioID);
 
   const [currentSimulationInputs, setCurrentSimulationInputs] = useState<
-    StdcmSimulationResult['input'] | undefined
+    StdcmSimulationInputs | undefined
   >(undefined);
 
-  const [interactedResultsElements, setInteractedResultsElements] = useState(false);
+  const [simulationsList, setSimulationsList] = useState<StdcmSimulation[]>([]);
+  const [selectedSimulationIndex, setSelectedSimulationIndex] = useState<number>(-1);
+  const [showSuccessBanner, setShowSuccessBanner] = useState<boolean>(false);
+  const [retainedSimulationIndex, setRetainedSimulationIndex] = useState(-1);
+  const dispatch = useAppDispatch();
+  const { resetStdcmConfig } = useOsrdConfActions() as StdcmConfSliceActions;
 
+  const selectedSimulation = simulationsList[selectedSimulationIndex];
+
+  // hide the simulation available card when the user interacts with the outputs.
+  const onInteractWithOutputs = () => setShowSuccessBanner(false);
+
+  // hide the simulation available card when the user interacts with the inputs.
+  // Even if currentSimulationInputs changed each times inputs changed, we can't use it to hide the card.
+  // Because it can change when the user change the selected simulation too.
+  const onInteractWithInputs = () => setShowSuccessBanner(false);
+
+  const retainSimulation = () => {
+    setRetainedSimulationIndex(selectedSimulationIndex);
+    onInteractWithOutputs();
+  };
+
+  const startNewQuery = () => {
+    setSimulationsList([]);
+    setSelectedSimulationIndex(-1);
+    setRetainedSimulationIndex(-1);
+    setShowSuccessBanner(false);
+    dispatch(resetStdcmConfig());
+  };
```

Comment:
Thank you for your feedback.

I understand your point about moving the functions closer to where they are used in StdcmResults. However, I chose to define these actions in the View component for the following reasons:

- **Centralization of Logic**: Having all the state logic in the View component ensures that the state management is centralized, making it easier **to track and debug**.

- **Avoid Passing Multiple State Variables**: These functions manipulate multiple state variables. Keeping the functions in View allows us to pass a single function as a prop instead of multiple state variables, which simplifies the props passing and keeps the components cleaner.

- **Consistency**: By centralizing complex actions, we ensure that the logic is not scattered and is easier to manage.

While moving the functions to `StdcmResults` might make them appear closer to where they are used, it would require passing multiple state variables through several levels, making the code harder to maintain.

I hope this clarifies my approach. What do you think?

Reply:
Ok, works for me :) Just wanted your reasons for this choice but I really like that the logic is grouped in this component !